packages/app/src/pages/transaction/index.tsx

```diff
@@ -0,0 +1,175 @@
+  return (
+    <Box whiteSpace="pre" overflow="scroll" color="gray" maxHeight="200px">
+      {JSON.stringify(json, null, 2)}
+    </Box>
+  );
+};
+
+const Textarea = styled.textarea`
+  position: absolute;
+  left: -999px;
+`;
+
+export const Transaction: React.FC = () => {
+  const location = useLocation();
+  const { identities } = useWallet();
+  const [requestState, setRequestState] = useState<RequestState | undefined>();
+  const [txHash, setTxHash] = useState('');
+  const [tx, setTx] = useState<StacksTransaction | null>(null);
+  const [copied, setCopied] = useState(false);
+
+  const getInputJSON = () => {
+    if (requestState) {
+      const { appDetails, publicKey, ...rest } = requestState;
+      return rest;
+    }
+    return {};
+  };
+
+  const tabs: Tab[] = [
+    {
+      title: 'Inputs',
+      content: <TabContent json={getInputJSON()} />,
+      key: 'inputs',
+    },
+    {
+      title: 'Outputs',
+      content: <TabContent json={tx} />,
+      key: 'outputs',
+    },
+    {
+      title: (
+        <>
+          View Source
+          <ExternalIcon display="inline-block" width="9px" ml={1} />
+        </>
+      ),
+      content: <span>src</span>,
+      key: 'source',
+    },
+  ];
+
+  useEffect(() => {
+    const requestToken = location.search
+      .substr(1)
+      .split('&')
+      .find(param => param.startsWith('request='))
+      ?.split('=')[1];
+    if (requestToken) {
+      const token = decodeToken(requestToken);
+      const reqState = (token.payload as unknown) as RequestState;
+      setRequestState(reqState);
+      console.log(reqState);
+      const [identity] = identities;
+      const args = reqState.functionArgs.map(arg => {
+        return new BufferCV(Buffer.from(arg));
+      });
+      const tx = makeContractCall(
+        reqState.contractAddress,
+        reqState.contractName,
+        reqState.functionName,
+        args,
+        new BN(0),
+        new BN(0), // TODO: actually track nonce
+        (identity.getSTXNode().privateKey as Buffer).toString('hex'),
+        TransactionVersion.Testnet
+      );
+      console.log(tx);
+      const serialized = tx.serialize().toString('hex');
+      setTxHash(serialized);
+      setTx(tx);
+      console.log(serialized);
+    }
+  }, []);
+
+  const handleButtonClick = () => {
+    const input: HTMLInputElement = document.querySelector('.hidden-tx-hash') as HTMLInputElement;
+    input.select();
+    input.setSelectionRange(0, 99999);
+    document.execCommand('copy');
+    setCopied(true);
+    document.getSelection()?.empty();
+    setTimeout(() => {
+      setCopied(false);
+    }, 1600);
+  };
+
+  return (
+    <>
+      <Screen>
+        {/* TODO: only show testnet banner if in testnet mode */}
+        <TestnetBanner />
+        <ScreenHeader
+          rightContent={
+            <Text textStyle="caption.small" color="gray" fontSize={0}>
+              {/* TODO: use RPC API to get balance */}
```

Comment:
Would this be RPC api, or the sidecar?

Reply:
🤷‍♂ , doesn't matter much to me. I think getting an account's status (and balance) would be pretty standard RPC.